modules/instrument_builder/php/instrument_builder.class.inc

```diff
@@ -35,32 +35,13 @@ class Instrument_Builder extends \NDB_Form
      * Tests that, when loading the Instrument builder module, some
      * text appears in the body.
      *
-     * @return bool
-     */
-    function _hasAccess()
-    {
-        return $this->_hasPerm('instrument_builder');
-    }
-    /**
-     * Wrapper to get the user object and check the permission.
-     *
-     * @param string $perm the value of perm
+     * @param \User $user The user whose access is being checked
      *
      * @return bool
      */
-    function _hasPerm($perm)
+    function _hasAccess(\User $user) : bool
     {
-        $user =& \User::singleton();
-        return $user->hasPermission($perm);
-    }
```

Comment:
Would it be a good idea to make a global `_hasAccess()` function that can be passed `$user` and an array of the `permissions` to be checked? It could potentially decouple a lot of this code.

Reply:
@HenriRabalais not sure that's it's necessary. If you need to check an array you could always do something like:

```php
$perms = [ 'permA, 'permB', ... ];
foreach ($perms as $p) { 
    if (!$user->hasPermission($p)) echo 'Forbidden!';
}
```

Reply:
There's already hasPermission in the User class, I'm not sure how having one that takes multiple permissions decouples it any more than a foreach loop.